serverless.yml

```diff
@@ -0,0 +1,112 @@
+
+
+service: users-api
+
+provider:
+  name: aws
+  runtime: nodejs10.x
+  iamRoleStatements:
```

Comment:
> What are your thoughts on best practices around role reuse?
> Do you tend to give each Lambda its distinct permissions or create IAM roles that can be reused across Lambdas?

I think this is another balancing act depending on the requirements.  For maximum security you would break down roles as finely as possible and restrict access to only what was absolutely necessary to that piece of compute.  This comes with overhead however and could become difficult to manage.  In many scenarios there may not be any risk in giving the `createUser` lambda the same permissions as the `updateUser`, `getUser` and `deleteUser` lambda.  Therefore you can reduce overhead by just using the same role.  

However I could envision scenarios where a user model may have a separate datastore(table) for NPPI data (social security number etc.).  This data may only be read perhaps by roles with elevated privileges ( a very specific service or admin).  This would seem an obvious reason to segregate the `readUsers` iam role from the `readUsersNppi` iam role.